src/App.tsx

```diff
@@ -1,55 +1,104 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  FC,
+  MouseEvent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import cn from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropDownContent } from './components/DropDownContent/DropDownContent';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = <T extends (...args: never[]) => void>(
+  func: T,
+  delay: number,
+) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: Parameters<T>): void => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
+
+export const App: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [query, setQuery] = useState('');
+  const [isSelectedQuery, setIsSelectedQuery] = useState(false);
+  const [selectedSlug, setSelectedSlug] = useState('');
+
+  useEffect(() => {
+    setPeoples(peopleFromServer);
+  }, []);
+
+  const queryWithDelay = useCallback(debounce(setIsSelectedQuery, 300), []);
```

Comment:
Here we will be creating function wich will set new state for `appliedQuery` with delay

```suggestion
  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
```